### PR TITLE
Fix stuck monitor recovery. Avoid DB related deadlocks. 

### LINKED ISSUE #1044: Skale admin doesn't restart schain monitoring if monitoring hangs
**Describe the bug**
In Skale Admin, there are two processes for each chain - one monitor is responsible for the status of contracts (creating chains, rotating ERC), and the second monitor checks the status of each Skaled container. Sometimes, the chain monitoring process hangs, ceases to initiate checks, and, in case of issues within the Skaled container, fails to perform the specified actions for recovery or restarting of such a Skaled.

**Preconditions:**
16 nodes 
8 active chains 
2 GETH nodes for 8 nodes each
Disabled auto_repair option in container config stream

**Versions:**
skalenetwork/admin:2.5.4-develop.1  

**To Reproduce**
Clear steps for reproduction are missing. 
To investigate, it is necessary for a failure to occur and to increase the log size for Skale-admin on QANet

**Expected behavior**
Skale admin should run 2 monitors after Geth nodes are synced

**Actual state:**
In some cases, skale admin initiates only contract monitoring, but does not restart the schain monitoring if it hangs. 

**Temporary workaround:**
Restart the Skale admin container manually

### PR DESCRIPTION
**Pull request by badrogger**
**Changes:**
1. Added stuck recovery mechanism for config/skaled schains pipelines. 
2. Switch to writing alive timestamp to a file instead of the db. 
3. Remove locks from db read/write operations. 

**Performance:**
1. No additional major computation or IO operations were introduced. 
 
**Tests:**
1. Added additional unit tests. 
4. Tested on the local network.

### CODE REVIEW

File: core/schains/monitor/main.py [hunk 1]
```diff
@@ -34,51 +32,61 @@
...
-from web.models.schain import SChainRecord
+from web.models.schain import SChainRecord, upsert_schain_record
 
 
 MIN_SCHAIN_MONITOR_SLEEP_INTERVAL = 20
 MAX_SCHAIN_MONITOR_SLEEP_INTERVAL = 40
 
-SKALED_PIPELINE_SLEEP = 2
-CONFIG_PIPELINE_SLEEP = 3
+STUCK_TIMEOUT = 60 * 60 * 2
```
Comment (DimaStebaev) [inline core/schains/monitor/main.py]: Better to specify time units in the name

File: core/schains/monitor/main.py [hunk 2]
```diff
@@ -167,36 +176,153 @@ def run_skaled_pipeline(
...
 
-def post_monitor_sleep():
-    schain_monitor_sleep = random.randint(
-        MIN_SCHAIN_MONITOR_SLEEP_INTERVAL, MAX_SCHAIN_MONITOR_SLEEP_INTERVAL
-    )
-    logger.info('Monitor iteration completed, sleeping for %d', schain_monitor_sleep)
-    time.sleep(schain_monitor_sleep)
+class SkaledTask(ITask):
+    NAME = 'skaled'
+    STUCK_TIMEOUT = 3600  # 1 hour
```
Comment (DimaStebaev) [inline core/schains/monitor/main.py]: STUCK_TIMEOUT_SECONDS

File: core/schains/monitor/tasks.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+from concurrent.futures import Future, ThreadPoolExecutor
+from typing import Callable
+
+from core.schains.process import ProcessReport
+
+
+logger = logging.getLogger(__name__)
+
+
+SLEEP_INTERVAL = 10
```
Comment (DimaStebaev) [inline core/schains/monitor/tasks.py]: SLEEP_INTERVAL_SECONDS?

File: core/schains/process_manager_helper.py [hunk 4]
```diff
@@ -1,92 +0,0 @@
-#   -*- coding: utf-8 -*-
```
Comment (DimaStebaev) [inline core/schains/process_manager_helper.py:1]: Remove the file?

**Review by DimaStebaev — DISMISSED** [review]

**Review by DmytroNazarenko — DISMISSED** [review]

File: core/schains/process.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+import pathlib
+import psutil
+
+from tools.configs.schains import SCHAINS_DIR_PATH
+from tools.helper import check_pid
+
+
+logger = logging.getLogger(__name__)
+
+TIMEOUT_COEFFICIENT = 2.2
```
Comment (dmytrotkk) [inline core/schains/process.py]: unused variable

File: core/schains/process.py [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+def get_schain_process_info(schain_name: str) -> Tuple[int | None, int | None]:
+    report = ProcessReport(schain_name)
+    if not ProcessReport(schain_name).is_exist():
+        return None, None
+    else:
+        return report.pid, report.ts
+
+
+class ProcessReport:
```
Comment (dmytrotkk) [inline core/schains/process.py:52]: does it make sense to move all basic functions, like save, save_tmp, update, move, etc, to the parent class that will be used for all json config files on the node?

**Review by dmytrotkk — CHANGES_REQUESTED** [review]

File: core/schains/process.py (hunk 5 above)
Comment (badrogger) [reply core/schains/process.py]: addressed

File: core/schains/monitor/tasks.py (hunk 3 above)
Comment (badrogger) [reply core/schains/monitor/tasks.py]: addressed

File: core/schains/monitor/main.py (hunk 2 above)
Comment (badrogger) [reply core/schains/monitor/main.py]: addressed

File: core/schains/monitor/main.py (hunk 1 above)
Comment (badrogger) [reply core/schains/monitor/main.py]: addressed

File: core/schains/process.py (hunk 6 above)
Comment (badrogger) [reply core/schains/process.py:52]: Suggest to do it in the next release.

**Pull request merged**